src/index.ts

```diff
@@ -49,17 +49,21 @@ export class Interference extends Error {
       writable: true,
     })
   }
+
+  public toJSON() {
+    return {
+      type: this.type,
+      message: this.message,
+      details: this.details,
+      statusCode: this.statusCode
+    }
+  }
 }
 
 export function isInterference(value: any): value is Interference {
   return typeof value === 'object' && value !== null ? value[symInterference] === true : false
 }
 
-export default function InterferenceFactory({
```

Comment:
This was the only way to be sure to pick only the needed props in JS  where you can pass any object to options. 

There is no performance problem since it is like creating a closure variable.

Reply:
There's another object destructuring inside the constructor, so the pick is still in place. By destructuring inside the `InterferenceFactory`, the engine is forced to extract data from the object, and then to create another one with those options, instead of just passing an object reference.

Reply:
You are right!